src/resolve/resolved-bus.c

```diff
@@ -2115,6 +2125,7 @@ static const sd_bus_vtable resolve_vtable[] = {
         SD_BUS_PROPERTY("DNSSECNegativeTrustAnchors", "as", bus_property_get_ntas, 0, 0),
         SD_BUS_PROPERTY("DNSStubListener", "s", bus_property_get_dns_stub_listener_mode, offsetof(Manager, dns_stub_listener_mode), 0),
         SD_BUS_PROPERTY("ResolvConfMode", "s", bus_property_get_resolv_conf_mode, 0, 0),
+        SD_BUS_PROPERTY("Monitor", "b", bus_property_get_monitor_enabled, 0, SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
```

Comment:
Please use `bus_property_get_bool` with `offsetof()`.

Reply:
BTW, is this necessary? I do not think so. I've not followed the above discussion, as it is already so long, but does this requested by someone?

Reply:
Yes, it was requested to indicate when this is active (and send a signal), given the privacy implications

Reply:
Hm? Sorry, I cannot follow. The property is not writable, and the flag will never updated...
Users can easily check that from the existence of the socket /run/systemd/resolve/io.systemd.Resolve.Monitor.

Reply:
It's a notification mechanism, dbus signal gets sent out

Reply:
Yeah, but why necessary? Is this for services possibly started before resolved? If so, why not to use inotify for the unix socket?

I'd like to add this when this becomes necessary.

Reply:
It is necessary for users to always know whether this is enabled or not

Reply:
Nah, then why not simply check the existence of the unix socket?

Reply:
Because that's a horrible interface - sockets can be stale, and don't mean anything, also access can be restricted but the check must be doable without privileges. Again, this was requested explicitly.

Reply:
Ah, I see. That makes sense for me. Thanks.

Reply:
@surajkrishnan14 Please ignore this thread except for the first one:
> Please use `bus_property_get_bool` with `offsetof()`.

Reply:
Updated to use `bus_property_get_bool` with `offsetof()`